test/dht_test.go

```diff
@@ -0,0 +1,122 @@
+package test
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-daemon/p2pclient"
+	pb "github.com/libp2p/go-libp2p-daemon/pb"
+)
+
+func TestDHTFindPeer(t *testing.T) {
+	daemon, client, closer := createMockDaemonClientPair(t)
```

Comment:
From here to [here](https://github.com/libp2p/go-libp2p-daemon/pull/17/files#diff-cce3ceb8f958169f0bab534fb7663872R27) is the same for each of the three test functions just with a `func()` passed in as a parameter, i.e:
```go
func setupMockDHT(tb *testing.TB, <insert good name here> func()) (daemon, []ids, client, closer, infochan)
```